eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -417,13 +394,113 @@ class Router(nodeParams: NodeParams, watcher: ActorRef) extends FSM[State, Data]
+      remote ! filter
+      stay
+
+    case Event(PeerRoutingMessage(remoteNodeId, routingMessage: HasChainHash), d) if routingMessage.chainHash != nodeParams.chainHash =>
+      sender ! TransportHandler.ReadAck(routingMessage)
+      log.warning("message {} for wrong chain {}, we're on {}", routingMessage, routingMessage.chainHash, nodeParams.chainHash)
+      stay
+
+    case Event(PeerRoutingMessage(remoteNodeId, u: ChannelUpdate), d) =>
+      sender ! TransportHandler.ReadAck(u)
+      log.debug("received channel update for shortChannelId={}", u.shortChannelId)
+      stay using handle(u, sender, d)
+
+    case Event(PeerRoutingMessage(remoteNodeId, c: ChannelAnnouncement), d) =>
+      log.debug("received channel announcement for shortChannelId={} nodeId1={} nodeId2={}", c.shortChannelId, c.nodeId1, c.nodeId2)
+      if (d.channels.contains(c.shortChannelId)) {
+        sender ! TransportHandler.ReadAck(c)
+        log.debug("ignoring {} (duplicate)", c)
+        stay
+      } else if (d.awaiting.contains(c)) {
+        sender ! TransportHandler.ReadAck(c)
+        log.debug("ignoring {} (being verified)", c)
+        // adding the sender to the list of origins so that we don't send back the same announcement to this peer later
+        val origins = d.awaiting(c) :+ sender
+        stay using d.copy(awaiting = d.awaiting + (c -> origins))
+      } else if (!Announcements.checkSigs(c)) {
+        sender ! TransportHandler.ReadAck(c)
+        log.warning("bad signature for announcement {}", c)
+        sender ! Error(Peer.CHANNELID_ZERO, "bad announcement sig!!!".getBytes())
+        stay
+      } else {
+        log.info("validating shortChannelId={}", c.shortChannelId)
+        watcher ! ValidateRequest(c)
+        // we don't acknowledge the message just yet
+        stay using d.copy(awaiting = d.awaiting + (c -> Seq(sender)))
+      }
+
+    case Event(PeerRoutingMessage(_, n: NodeAnnouncement), d: Data) =>
+      sender ! TransportHandler.ReadAck(n)
+      log.debug("received node announcement for nodeId={}", n.nodeId)
+      stay using handle(n, sender, d)
+
+    case Event(PeerRoutingMessage(_, routingMessage@QueryChannelRange(chainHash, firstBlockNum, numberOfBlocks)), d) =>
+      sender ! TransportHandler.ReadAck(routingMessage)
+      log.info("received query_channel_range={}", routingMessage)
+      // sort channel ids and keep the ones which are in [firstBlockNum, firstBlockNum + numberOfBlocks]
+      val shortChannelIds: SortedSet[ShortChannelId] = d.channels.keySet.filter(keep(firstBlockNum, numberOfBlocks, _, d.channels, d.updates))
+      if (shortChannelIds.isEmpty) {
+        // tell them we have nothing
+        log.info("sending back empty reply_channel_range for range=({}, {})", firstBlockNum, numberOfBlocks)
+        sender ! ReplyChannelRange(chainHash, firstBlockNum, numberOfBlocks, 1, BinaryData("00"))
```

Comment:
Why isn't the `00` encoding done in the `ChannelRangeQueries.encodeShortChannelIds`? That would save us a special case